### DISCUSSION TITLE
Ranking with context-aware models?

### DISCUSSION THREAD

**Discussion by deklanw** [RUCAIBox/RecBole#672] [Q&A] [answered]
I know that many of these CTR prediction models are trained in a pointwise binary classification setting. But, in principle they can be applied to ranking since you can just sample negatives and assign `0` labels.

To give an explicit example, I want to try DCN with `ml-100k` treating all rated movies as positive and then sample, e.g., 2 negative examples for each positive (instead of thresholding ratings and treating low values as negatives). With this setup I'd like to use precision, ndcg, etc as metrics.

Is this supported in RecBole?

**Comment by batmanfly** [comment, marked as answer]
@deklanw We have ready one version that includes this feature, which is to be released soon (within three days ). =)

The feature should be included in the first release. While, we ignored it previously. It is definitely important.

Our team will update this page by including the code reference once it is officially released. 

Thanks!
